runtime/src/tests.rs

```diff
@@ -100,3 +168,204 @@ fn pallet_poe_availability() {
         // just checking code builds, hence the pallet is available to the runtime
     });
 }
+
+mod pallets_interact {
+    use super::*;
+
+    #[test]
+    fn session_notifies_staking() {
+        new_test_ext().execute_with(|| {
+            let pre_staking_session = Staking::current_planned_session();
+            Session::rotate_session();
+            let post_staking_session = Staking::current_planned_session();
+            assert_eq!(pre_staking_session + 1, post_staking_session);
+        });
+    }
+
+    #[test]
+    fn authorship_is_configured_with_aura() {
+        new_test_ext().execute_with(|| {
+            let block_number = 1;
+            let slot = Slot::from(SLOT_ID);
+            let pre_digest = Digest {
+                logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+            };
+            System::reset_events();
+            System::initialize(&block_number, &Default::default(), &pre_digest);
+
+            assert!(
+                Authorship::author()
+                    == Some(AccountId32::new(
+                        testsfixtures::SAMPLE_USERS[AURA_AUTHOR_ID as usize]
+                            .raw_account
+                            .into()
+                    ))
+            );
+        });
+    }
+
+    // Check that Authorship calls back on ImOnline
+    #[test]
+    fn authorship_notifies_imonline() {
+        new_test_ext().execute_with(|| {
+            let block_number = 1;
+            let slot = Slot::from(SLOT_ID);
+            let pre_digest = Digest {
+                logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+            };
+            System::reset_events();
+            System::initialize(&block_number, &Default::default(), &pre_digest);
+
+            assert!(!ImOnline::is_online(AURA_AUTHOR_ID));
+            Authorship::on_initialize(block_number);
+            assert!(ImOnline::is_online(AURA_AUTHOR_ID));
+        });
+    }
+
+    #[test]
+    fn authorship_notifies_staking() {
+        new_test_ext().execute_with(|| {
+            let block_number = 1;
+            let slot = Slot::from(SLOT_ID);
+            let pre_digest = Digest {
+                logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+            };
+            System::reset_events();
+            System::initialize(&block_number, &Default::default(), &pre_digest);
+
+            // Before authoring a block, no points have been given in the active era
+            assert!(
+                Staking::eras_reward_points(Staking::active_era().expect("No active era").index)
+                    .total
+                    == 0
+            );
+
```

Comment:
As general question...  Why do you often check prerequirment contitions in your tests? Almost of the time are trivial true... I'm trying to avoid them and write this checks only when I really need them, but is just mater of tasting if you prefer this form it's fime for me.

Reply:
I agree it's not a big deal, but I find it helps clarifying the flow of the test and makes the expected precondition stand out.
Also, it should be safer in case of future breaking modifications (even though unlikely)?